rllib/core/rl_trainer/trainer_runner_config.py

```diff
@@ -0,0 +1,171 @@
+        self.action_space = None
+        self.model_config = None
+
+        # `self.trainer()`
+        self.trainer_class = None
+        self.eager_tracing = True
+        self.optimizer_config = None
+
+        # `self.resources()`
+        self.num_gpus = 0
+        self.fake_gpus = False
+
+        # `self.testing()`
+        self._in_test = False
+
+    def validate(self) -> None:
+
+        if self.module_class is None and self.module_obj is None:
+            raise ValueError(
+                "Cannot initialize an RLTrainer without an RLModule. Please provide "
+                "the RLModule class with .module(module_class=MyModuleClass) or "
+                "an RLModule instance with .module(module=MyModuleInstance)."
+            )
+
+        if self.module_class is not None:
+            if self.observation_space is None:
+                raise ValueError(
+                    "Must provide observation_space for RLModule when RLModule class "
+                    "is provided. Use .module(observation_space=MySpace)."
+                )
+            if self.action_space is None:
+                raise ValueError(
+                    "Must provide action_space for RLModule when RLModule class "
+                    "is provided. Use .module(action_space=MySpace)."
+                )
+            if self.model_config is None:
+                raise ValueError(
+                    "Must provide model_config for RLModule when RLModule class "
+                    "is provided. Use .module(model_config=MyConfig)."
+                )
+
+        if self.trainer_class is None:
+            raise ValueError(
+                "Cannot initialize an RLTrainer without an RLTrainer. Please provide "
+                "the RLTrainer class with .trainer(trainer_class=MyTrainerClass)."
+            )
+
+        if self.optimizer_config is None:
+            # get the default optimizer config if it's not provided
+            # TODO (Kourosh): Change the optimizer config to a dataclass object.
+            self.optimizer_config = {"lr": 1e-3}
+
+        if self.fake_gpus and self.num_gpus <= 0:
+            raise ValueError("If fake_gpus is True, num_gpus must be greater than 0.")
+
+    def build(self) -> TrainerRunner:
+        self.validate()
+        # TODO (Kourosh): What should be scaling_config? it's not clear what
+        # should be passed in as trainer_config and what will be inferred
+        return self.trainer_runner_class(
+            trainer_class=self.trainer_class,
+            trainer_config={
+                "module_class": self.module_class,
+                "module_kwargs": {
+                    "observation_space": self.observation_space,
+                    "action_space": self.action_space,
+                    "model_config": self.model_config,
+                },
+                # TODO (Kourosh): should this be inferred inside the constructor?
+                "distributed": self.num_gpus > 1,
```

Comment:
Training on multiple cpus is equivalent to `self.num_gpus > 1` but `fake_gpus=True`. Even in that case num_gpus > 1 is equivalent to distributing the module to make sure the gradients on synchronized.

Reply:
ok, let's keep this gocha in mind.
I don't know if it's intuitive to configure things this way.
like instead of ``num_gpus > 1`` and ``fake_gpus=True``, I would imagine we just set something like ``num_training_workers>1``

Reply:
This is how we do it today. No body does multi core training in deep rl man :-)) these are the parameters we had from algo config already and it is how they were documented before.

Reply:
yeah, understand. I am just saying our current state doesn't seem ideal :)